src/main/java/duke/EventHandler.java

```diff
@@ -0,0 +1,144 @@
+package duke;
+
+/**
+ * Class responsible for handling all events in Duke
+ */
+public class EventHandler {
+
+    private TaskList taskList;
+    private UserInterface userInterface;
+    private GuiUserInterface guiUserInterface;
+    private Storage storage;
+
+    /**
+     * Contructor for EventHandler.
+     *
+     * @param taskList      TaskList for EventHandler to handle.
+     * @param userInterface UserInterface that EventHandler uses.
+     */
+    public EventHandler(TaskList taskList, UserInterface userInterface,
+                        GuiUserInterface guiUserInterface, Storage storage) {
+        this.taskList = taskList;
+        this.userInterface = userInterface;
+        this.guiUserInterface = guiUserInterface;
+        this.storage = storage;
+    }
+
+    /**
+     * Adds a ToDo into taskList.
+     *
+     * @param input User input
+     * @throws DukeNoDescriptionException Thrown when description is empty.
+     */
+    public String addTodo(String input) throws DukeNoDescriptionException {
+        if (input.length() == 4) {
+            throw new DukeNoDescriptionException();
+        }
+        String description = input.substring(5);
+        ToDo tempTask = new ToDo(description);
+        this.taskList.add(tempTask);
+//        userInterface.addTaskMessage(tempTask);
+        storage.save();
+        return guiUserInterface.addTaskMessage(tempTask);
+    }
+
+    /**
+     * Adds a Deadline into taskList.
+     *
+     * @param input User input
+     * @throws DukeNoDescriptionException Thrown when description is empty.
+     */
+    public String addDeadline(String input) throws DukeNoDescriptionException {
+        if (input.length() == 8) {
+            throw new DukeNoDescriptionException();
+        }
+
+        int endAt = input.indexOf("/");
```

Comment:
Maybe endIndex can be a better choice here?

Reply:
I agree with this. endIndex sounds more descriptive imo.